### PR TITLE
Add comments in ShaderLang.h

### PR DESCRIPTION
**Pull request by realitix**
`ShCompile` function is not so easy to use and need more explanations.

### CODE REVIEW

**Comment by johnkslang** [comment]
This path is deprecated. Is this a legacy use?  Hopefully, any new use would be through TShader.parse(), and hopefully that interface makes more sense.  See comments in the middle of ShaderLang.h.

File: glslang/Public/ShaderLang.h [hunk 1]
```diff
@@ -190,13 +190,16 @@ SH_IMPORT_EXPORT void ShDestruct(ShHandle);
...
+//
 SH_IMPORT_EXPORT int ShCompile(
-    const ShHandle,
-    const char* const shaderStrings[],
-    const int numStrings,
-    const int* lengths,
-    const EShOptimizationLevel,
-    const TBuiltInResource *resources,
+    const ShHandle,                      // compiler object
+    const char* const shaderStrings[],   // shader splitted in numStrings part
```
Comment (johnkslang) [inline glslang/Public/ShaderLang.h]: -> shader split into 'numString's parts

File: glslang/Public/ShaderLang.h [hunk 2]
```diff
@@ -190,13 +190,16 @@ SH_IMPORT_EXPORT void ShDestruct(ShHandle);
...
+    const int numStrings,                // length of shaderStrings
```
Comment (johnkslang) [inline glslang/Public/ShaderLang.h]: -> number of shader strings

(The variable name was more accurate.)

File: glslang/Public/ShaderLang.h [hunk 3]
```diff
@@ -190,13 +190,16 @@ SH_IMPORT_EXPORT void ShDestruct(ShHandle);
...
+    const int* lengths,                  // list of lengths (corresponding to each shaderStrings), can be NULL to use \0
```
Comment (johnkslang) [inline glslang/Public/ShaderLang.h]: NULL means a normal 0-terminated C string was passed in

**Comment by realitix** [comment]
@johnkslang I am doing a Python wrapper with Cffi so I need C wrapper. But after 2 days, I have realized that this C API is not usable. The C header can't compile, I don't manage to compile a single glsl shader to Spir-V... So I now use libshaderc which provide good C headers.
Maybe this C header must be removed and the documentation updated to redirect to shaderc for C header.
What do you think about that?

**Comment by johnkslang** [comment]
@realitix I created issue #751 for this, thanks.

**Pull request closed without merging**